workers/agent/agent.go

```diff
@@ -454,12 +472,54 @@ func (a *Agent) scheduleWork() {
 
 		// Let us try to start the plugin and receive all implemented jobs
 		if err = a.scheduler.SetPipelineJobs(pipeline); err != nil {
-			gaia.Cfg.Logger.Error("cannot get pipeline jobs", "error", err.Error(), "pipelinerun", pipelineRunPB)
-			reschedulePipeline()
-			return
+			if !strings.Contains(err.Error(), "exec format error") {
+				gaia.Cfg.Logger.Error("cannot get pipeline jobs", "error", err.Error(), "pipelinerun", pipelineRunPB)
+				reschedulePipeline()
+				return
+			}
+			gaia.Cfg.Logger.Info("pipeline in a different format than worker; attempting to rebuild...")
+			// Try rebuilding the pipeline...
+			if err := os.Remove(pipelineFullPath); err != nil {
+				gaia.Cfg.Logger.Error("failed to remove pipeline binary", "error", err.Error(), "pipelinerun", pipelineRunPB)
+				reschedulePipeline()
+				return
+			}
+			err = a.rebuildWorkerBinary(ctx, pipeline)
+			if err != nil {
+				gaia.Cfg.Logger.Error("failed to rebuild pipeline for worker", "error", err.Error(), "pipelinerun", pipelineRunPB)
+				reschedulePipeline()
+				return
+			}
+
+			workerSHA256Sum, err := filehelper.GetSHA256Sum(pipelineFullPath)
+			if err != nil {
+				gaia.Cfg.Logger.Error("failed to determine SHA256Sum of pipeline file", "error", err.Error(), "pipelinerun", pipelineRunPB)
+				reschedulePipeline()
+				return
+			}
+
+			uID := strconv.Itoa(pipelineRun.PipelineID)
+			shaPair := gaia.SHAPair{
+				Original: pipelineSHA256SUM,
+				Worker:   workerSHA256Sum,
+				UniqueID: uID,
+			}
+
+			err = db.UpsertSHAPair(shaPair)
```

Comment:
MemDB is only in-memory. What happens when a worker has been restarted?

Reply:
Good question. It will disappear and would have to be added again. So if a worker stops it will re-compile everything after startup. Do you think that's reasonable thing to do? Or should we rather put things into the master's db?

Reply:
> Or should we rather put things into the master's db?

No, that is not an option. But we could put it in the worker db, couldn't we? 😄

Reply:
@michelvocks Wait, uh... um... that's what I thought the memdb is? I might be confused here a bit. :D Would you mind clearing this up? Which one is the worker db? :D

Reply:
We also use the boltdb (storage) on the worker side. MemDB is just a cache right before the storage to speedup certain things

Reply:
Ah! Gotcha. Okay. I'll switch to boltdb. :)

Reply:
Oh dang. I see now. This will make things a bit easier actually and maybe remove some added code. :)